src/Mvc/Mvc.Core/src/ApplicationParts/RelatedAssemblyAttribute.cs

```diff
@@ -66,7 +66,7 @@ public static IReadOnlyList<Assembly> GetRelatedAssemblies(Assembly assembly, bo
 
             // MVC will specifically look for related parts in the same physical directory as the assembly.
             // No-op if the assembly does not have a location.
-            if (assembly.IsDynamic || string.IsNullOrEmpty(assembly.CodeBase))
+            if (assembly.IsDynamic || string.IsNullOrEmpty(assembly.Location))
```

Comment:
Is this right? Will this not fail with the new publish single file work?

Reply:
I'm not entirely sure how that works, but we should tackle that separately. This doesn't make the code more incorrect than before.

Reply:
That's fair, can we file an issue to track that work?

Reply:
Why check IsDynamic?

Reply:
We had issues where users had controllers in dynamically generated assemblies.

Reply:
But what does that check have to do with anything here?

Reply:
* The way you add the controller assembly is by registering it as an application part. 
* MVC will attempt to look up related parts by using the assembly's location. 
* Accessing the location property on a dynamically generated assembly throws.

If the stars align, MVC will throw. Does that help?